internal/common.go

```diff
@@ -0,0 +1,12 @@
+package internal
```

Comment:
Adding a common at this level to share types and utility functions between the domain model and data model (and those layers in general where it makes sense)